hudi-common/src/main/java/org/apache/hudi/common/config/HoodieMetadataConfig.java

```diff
@@ -316,6 +316,13 @@ public final class HoodieMetadataConfig extends HoodieConfig {
       .withDocumentation("Initializes the metadata table by reading from the file system when the table is first created. Enabled by default. "
           + "Warning: This should only be disabled when manually constructing the metadata table outside of typical Hudi writer flows.");
 
+  public static final ConfigProperty<Boolean> FUNCTIONAL_INDEX_ENABLE_PROP = ConfigProperty
+      .key(METADATA_PREFIX + ".functional.index.enable")
```

Comment:
the equivalent record index config is formatted as `.record.index.enable`

So that matches this config. However, for the functional index, the configs are 
`.index.functional.file.group.count` `.index.functional.parallelism`

and the equivalent partition stats config is `.index.partition.stats.enable`

I think since the record index configs are 0.14 but the other index configs are 1.0, we should probably just change them all to mach the record index ones. So `.partition.stats.index.enable` `.functional.index.file.group.count` `.partition.index.stats.enable` maybe we can create some more static prefix strings?

Reply:
But I think there is an argument that the index part should go first. So all the mdt index configs have the same prefix. WDYT?

Reply:
You raise a good point. I simply named the key that way because we keep calling functional index. I noticed that it is only the record index which deviates from the standard,e.g. bloom filter - `.index.bloom.filter.enable`, colstats - `.index.column.stats.enable`. So, I am going to rename it to `.index.functional.enable`. I can take up renaming record index enable key and use alias for compatibility in a followup.